include/pressio/ops/kokkos/ops_clone.hpp

```diff
@@ -53,30 +53,20 @@ namespace pressio{ namespace ops{
 
 template <typename T>
 ::pressio::mpl::enable_if_t<
-  (::pressio::is_native_container_kokkos<T>::value
-  or ::pressio::is_expression_acting_on_kokkos<T>::value)
-  and ::pressio::Traits<T>::rank == 1,
-  T
+  // common clone constraints
+    (::pressio::Traits<T>::rank == 1
+  || ::pressio::Traits<T>::rank == 2)
+  // TPL/container specific
+  && ::pressio::is_native_container_kokkos<T>::value,
+  decltype(Kokkos::create_mirror(std::declval<
+    const T &
+    >()))
   >
 clone(const T & clonable)
 {
-  T r(clonable.label(), clonable.extent(0));
-  Kokkos::deep_copy(r, clonable);
-  return r;
-}
-
-template <typename T>
-::pressio::mpl::enable_if_t<
-  (::pressio::is_native_container_kokkos<T>::value
-  or ::pressio::is_expression_acting_on_kokkos<T>::value)
-  and ::pressio::Traits<T>::rank == 2,
-  T
-  >
-clone(const T & clonable)
-{
-  T r(clonable.label(), clonable.extent(0), clonable.extent(1));
-  Kokkos::deep_copy(r, clonable);
-  return r;
+  auto mirror = Kokkos::create_mirror(clonable);
```

Comment:
@mzuzek why did you change the implementation? 
This is not correct: create_mirror does not clone anything, it is a clone just because you are doing things on host. Please use the implementation I had before, maybe with the minor fix for the label:
```
  T r(clonable.label()+"_clone", clonable.extent(0));
  Kokkos::deep_copy(r, clonable);
  return r;
```

Reply:
thanks, fixed on 07f5020901b5c7cfbb60039cdfd95c84b1b2cf42